HIL integration for Gazebo

arwagoner opened the pull request:
This PR integrates HIL for Gazebo. Gazebo listens to a serial port for a connection with the FCU and establishes a UDP proxy to QGC.

jgoppert committed:
Added gps noise and delay model.

Commit:
Enable Video Streaming by default

Commit:
Convert Frame Format from RGB to YUV

QGC receiving the video stream doesnt support RGB format. Camera
in Gazebo doesnt source YUV frames. It is therefore required to
do the color conversion from RGB to YUV.

Commit:
Video Widget to enable/disable videostreaming

To help users enable or disable video streaming run time

Commit:
Start/Stop Video Streaming runtime

Based on user selection, video streaming can be started or stopped. User
chooses by pressing GUI overlay button. Selection is published on a topic.
Video streaming plugin receives the selection over the subscribed topic and
then starts or stops the streaming based on its current state.

arwagoner committed:
HIL integration for Gazebo.

arwagoner committed:
HIL integration for Gazebo.

TSC21 commented on include/gazebo_mavlink_interface.h:
```diff
@@ -17,6 +17,23 @@
+#include <cassert>
+#include <stdexcept>
+#include <boost/asio.hpp>
+#include <boost/bind.hpp>
```
double include of `bind.hpp` - look at line 38.

arwagoner committed:
Removed extra include statement.

arwagoner committed:
Removed extra include statement.

TSC21 commented on models/rotors_description/urdf/iris_base.xacro:
```diff
@@ -7,6 +7,13 @@
   <xacro:arg name='name' default='iris' />
   <xacro:arg name='mavlink_addr' default='INADDR_ANY' />
   <xacro:arg name='mavlink_udp_port' default='14560' />
+  <xacro:arg name='serial_enabled' default='false' />
```
we are missing these parameters on the SDF of the other files.

TSC21 commented on include/gazebo_mavlink_interface.h:
```diff
@@ -105,7 +141,19 @@ class GazeboMavlinkInterface : public ModelPlugin {
+        device_(kDefaultDevice),
+        baudrate_(kDefaultBaudRate),
+        hil_mode_(false),
+        hil_state_level_(true)
```
`hil_state_level_(false)`?

arwagoner replied:
Yes. I agree.

arwagoner committed:
Added params to sdf for all vehicle.

arwagoner committed:
Added params to sdf for all vehicle.

arwagoner committed:
Added params to sdf for all vehicle.

arwagoner committed:
hil_state_mode default changed to false

arwagoner committed:
hil_state_mode default changed to false

arwagoner committed:
hil_state_mode default changed to false

arwagoner committed:
hil_state_mode default changed to false

LorenzMeier committed:
GPS Plugin: Tune down noise factors, fix build on Gazebo 8

Gazebo has its own GPS plugin (which we should use in the future). The noise was also configured as way too strong and completely unrealistic, resulting in weird in-flight behaviour.

LorenzMeier committed:
MAVLink: Update MAVLink search script

LorenzMeier committed:
Video streaming: Fix compilation and execution on Gazebo8

LorenzMeier committed:
CMAKE: Do not use unset variables

LorenzMeier committed:
Temporarily disable GUI plugin

TSC21 committed:
find user built headers in /usr/include or /usr/local/include

dagar committed:
gstreamer plugin temporarily disable

dagar committed:
initial travis-ci configuration (#136)

TSC21 committed:
README: add Travis CI build status token

TSC21 committed:
GPS noise and delay sim: minor fixes

TSC21 committed:
update OpticalFlow submodule

dagar committed:
standard_vtol fix gpsSubTopic tag

dagar committed:
hippocampus fix gpsSubTopic tag

dagar committed:
solo fix gpsSubTopic tag

dagar committed:
tailsitter fix gpsSubTopic tag

anassinator committed:
Fix inverted magnetic declination (#96)

LorenzMeier committed:
Throttle GPS updates to 5 Hz

TSC21 committed:
gazebo_gps_plugin: set GPS delay to 120ms

TSC21 committed:
fix groundtruth data propagation

TSC21 committed:
gazebo_gps: add groundtruth specific callback and data msg propagation

TSC21 committed:
plugins: style fix gps and mavlink_interface

arwagoner committed:
Added HITL to title.

arwagoner committed:
Added HITL to title.

arwagoner committed:
Added HITL to title.

arwagoner committed:
Added HITL to title.

TSC21 committed:
some minor rebase fixes

arwagoner committed:
Fixed conflicts

arwagoner committed:
Fixed conflicts

arwagoner committed:
Fixing conflicts

arwagoner committed:
Merge branch 'HIL_integration' of github.com:arwagoner/sitl_gazebo into HIL_integration

arwagoner committed:
Minor fixes
